src/flatpak-spawn.c

```diff
@@ -495,6 +506,10 @@ check_portal_supports (const char *option, guint32 supports_needed)
   if ((supports & supports_needed) != supports_needed)
     {
       g_printerr ("--%s not supported by host portal\n", option);
+
+      if (supports_needed == FLATPAK_SPAWN_SUPPORT_FLAGS_EXPOSE_PIDS)
```

Comment:
This needs a bitwise check, not ==

Reply:
What bitwise check do you have in mind? At the moment there's only one bit implemented (`EXPOSE_PIDS` and `SHARE_PIDS` are aliases for each other), so the practical effect is the same regardless; the only thing we'd get from having a bitwise check is nudging future maintainers in the right direction when they add a second bit.

So... if we later add an option `--foobar` that requires both `EXPOSE_PIDS` and a second bit `RETICULATED_SPLINES`, what UX do you want it to have?

If I did for example

```suggestion
      if (supports_needed & FLATPAK_SPAWN_SUPPORT_FLAGS_EXPOSE_PIDS)
```

then we'd print:

```
--foobar not supported by host portal

This feature requires Flatpak to be using a bubblewrap (bwrap) executable
that is not setuid root.
...
```

which seems maybe misleading - it needs bwrap to not be setuid root, but it also needs whatever system property (sysctl? kernel configuration?) lets us implement `RETICULATED_SPLINES`, and we're not talking about that here?

Reply:
What I was thinking here is that by using `==`, we'll have this UX:

* checking for `EXPOSE_PIDS` only -> print the long explanation
* checking for `EXPOSE_PIDS|RETICULATED_SPLINES` -> degrade to only printing the short version as we do now, and users can do their own research to find out how to make `--foobar` work
* checking for `RETICULATED_SPLINES` only -> the same

Reply:
Well, supposedly we would then also have a bitwise check for RETICULATED_SPLINES and print a special warning for that, and then --foobar would print both.

But you're right that it's not clear that doing the more complex check is better, so maybe we should leave it as is until we actually have more bits.